test/address.test.js

```diff
@@ -198,13 +199,97 @@ define(function(require) {
           .query('a', 'a=b')
           .header('accept','application/json')
           .body({hello:'world!'})
+          .timeout(60)
           .on('done', cb)
 
         update()
 
         web.req.should.have.been.calledOnce
         web.req.should.have.been.calledWith(req)
+        cb.should.have.been.calledOnce
+      })
+
+      it('should call configured request and callback with the default timeout if no value is provided', function() {
+        var cb = sinon.spy()
+          , req = {
+            uri : '/wibble/123'
+          , method : 'send'
+          , headers : {
+              accept : 'application/json'
+            }
+          , body : undefined
+          , context: undefined
+          , origin: undefined
+          , timeout: 30
+          }
+
+        var update = address('/wibble/{id}')
+          .param('id', '123')
+          .method('send')
+          .header('accept','application/json')
+          .timeout()
+          .on('done', cb)
+
+        update()
+
+        web.req.should.have.been.calledOnce
+        web.req.should.have.been.calledWith(req)
+        cb.should.have.been.calledOnce
+      })
+
+      it('should call configured request and callback with the default timeout if negative value is provided', function() {
+        var cb = sinon.spy()
+          , req = {
+            uri : '/wibble/123'
+          , method : 'send'
+          , headers : {
+              accept : 'application/json'
+            }
+          , body : undefined
+          , context: undefined
+          , origin: undefined
+          , timeout: 30
+          }
+
+        var update = address('/wibble/{id}')
+          .param('id', '123')
+          .method('send')
+          .header('accept','application/json')
+          .timeout(-10)
+          .on('done', cb)
+
+        update()
+
+        web.req.should.have.been.calledOnce
+        web.req.should.have.been.calledWith(req)
+        cb.should.have.been.calledOnce
+      })
+
+      it('should call configured request and callback with the default timeout if null value is provided', function() {
```

Comment:
No, again, we should throw an error. Invalid values shouldn't transform to some other value.